### ISSUE TITLE
"Allow Multiple Boxes for Shipping" setting "yes" can not be saved

### ISSUE THREAD

**Issue by kmrant** [magento/magento2#13342]
### Preconditions

1. Magento 2.4-develop.

### Steps to reproduce

1. Open product for editing: Products - catalog - click a product.
2. Click "Advanced inventory".
3. Change "Allow Multiple Boxes for Shipping" setting to "yes".
4. Save.

### Expected result

1. Setting should now appear as yes.

### Actual result

1. Setting is still no.

**Comment by magento-engcom-team** [comment]
@kmrant, thank you for your report.
We've acknowledged the issue and added to our backlog.

**Comment by enriquei4** [comment]
Hi @magento-engcom-team and @kmrant I think that this is the expected result because "Allow Multiple Boxes for Shipping" (is_decimal_divided). Depends of " Qty Uses Decimals" (is_qty_decimal).
You can see at: https://github.com/magento/magento2/blob/6ce8b488e38f26773d7f05a2d93ecfe9746b40d2/app/code/Magento/Catalog/Controller/Adminhtml/Product/Initialization/StockDataFilter.php#L67 

Then is required enable  "Qty Uses Decimals" for use "Allow Multiple Boxes for Shipping" in advanced inventory

**Comment by kmrant** [comment]
Hi,

You are correct. It works if I choose "Qty Uses Decimals" yes. It would 
be good to have mention about this next to the "Allow Multiple Boxes for 
Shipping".

--
Best Regards,
Kimmo Räntilä
SatShop.fi
Suomen satelliittikauppa Oy

25.1.2018, 18.04, enriquei4 kirjoitti:

**Comment by kmrant** [comment]
Hi,

But I still can't create a shipment with 2 packages for 1 product. I 
tried to change number of products to 0,5 or 0.5 but it's still 1 and 
it's not possible to add same product to another package too.

--
Best Regards,
Kimmo Räntilä
SatShop.fi
Suomen satelliittikauppa Oy

25.1.2018, 18.04, enriquei4 kirjoitti:

**Comment by enriquei4** [comment]
I think that this feature works if you have 2 items (or more) and want to send it separatelly, not to split 1 product in different boxes. @magento-engcom-team  right?

**Comment by enriquei4** [comment]
Also @magento-engcom-team, could I create a hotfix to make the field "Allow Multiple Boxes for Shipping" depending on "Qty Uses Decimals" in admin?

**Comment by pemann** [comment]
@magento-engcom-team why "Allow multiple boxes for shipping" is depending on "qty uses decimals"?

I'm selling big products which are shipped in multiple packages, but there is no point in selling partial products. So the qty should be integer, but I need to split it into multiple packages when shipping. Currently this cannot be done as it should be enabling only the "Allow multiple boxes for shipping", but leaving "qty uses decimals" to false. Futhermore at order shipping grid the input does not even read the multiple boxes setting. See:
https://github.com/magento/magento2/blob/2.2-develop/app/code/Magento/Shipping/view/adminhtml/templates/order/packaging/grid.phtml#L91

I think these two settings should be separated.

**Comment by magento-engcom-team** [comment]
:white_check_mark: Confirmed by @engcom-Delta 
 Thank you for verifying the issue. Based on the provided information internal tickets `MC-30157` were created

**Issue Available**: @engcom-Delta, _You will be automatically unassigned. Contributors/Maintainers can claim this issue to continue. To reclaim and continue work, reassign the ticket to yourself._

**Comment by engcom-Alfa** [comment]
![Peek 2020-08-03 17-38](https://user-images.githubusercontent.com/51679138/89194712-61cce600-d5b0-11ea-9acf-03e3808e1799.gif)

**Comment by magento-engcom-team** [comment]
Hi @kmrant. 

Thank you for your report and collaboration!

The related internal Jira ticket `MC-30157` was closed as `Fixed`. 

The fix will be available with the upcoming `2.4.3` release.

**Comment by juharintanen** [comment]
> 
> 
> Hi @kmrant.
> 
> Thank you for your report and collaboration!
> 
> The related internal Jira ticket `MC-30157` was closed as `Fixed`.
> 
> The fix will be available with the upcoming `2.4.3` release.

Is the fix that that its clearly visible that Allow Multiple Boxes for Shipping depends on decimal qty
OR are we now allowed to create multiple packages for a single product?

**Comment by nikoelgatito** [comment]
I believe that this issue should be reopened, as the behavior where the `is_decimal_divided` attribute is depending on the `is_qty_decimal` attribute does not make sense.

If a product can be sold in fractions, it should also be possible to ship it in fraction following common logic. The `is_decimal_divided` attribute should instead allow to divide a product that is only sold in increment of 1 into fractions for shipping, where it wouldn't be possible otherwise, as mentioned by @pemann.

Since it is already possible to ship a product having its `is_qty_decimal` attribute value set to true by default, the `is_decimal_divided` attribute does not serve any useful purpose.

My understanding is that these should be completely independent, where `is_decimal_divided` should control the shipping behavior only, under all circumstances.

An example of where this behavior would be limiting includes situations where merchants would create any kind of product bundle using the simple product type for convenience, since it does not require creating multiple product SKUs. When time comes for shipping, it is sometimes necessary to split the bundle into different parts, which is not currently possible.

After searching for `is_decimal_divided` in the different Magento modules and proceeding to a few tests, it seems that the `is_decimal_divided` attribute value is not actually used anywhere and does not have any impact on shipping.

It would be useful if it could override the condition setting the qty-decimal class in the file below, allowing to use decimals for shipping if the `is_decimal_divided` attribute is set to true.

**File:** app/code/Magento/Shipping/view/adminhtml/templates/order/packaging/grid.phtml
**Line:** 109
```
<?php if ($item->getOrderItem()->getIsQtyDecimal()): ?>
    qty-decimal
<?php endif ?>">
```

With the current behavior, it is literally impossible to sell a product that is not using a decimal qty and ship it in multiple boxes, as "Allow Multiple Boxes for Shipping" implies. Magento does not currently allow such a simple operation. This is an important issue, as this is a scenario that is commonly faced by merchants. In my opinion, this is one of many reasons why merchants are turning to alternative solutions for shipping instead of relying on the built-in functionalities.

**Fields Definition:**
Qty Uses Decimals = `is_qty_decimal`
Allow Multiple Boxes for Shipping = `is_decimal_divided`
